watcher/lib/grafana/grafana.go

```diff
@@ -90,3 +92,21 @@ type CreateDashboardRequest struct {
 	// Overwrite is whether to overwrite existing dashboard with newer version or with same dashboard title
 	Overwrite bool `json:"overwrite"`
 }
+
+// DeleteDashboard deletes dashboard
+// dashboard name is obtained from dashboard data field 'title'
+func (c *Client) DeleteDashboard(data string) error {
+	// dashboard data should be a valid JSON
+	var dashboardJSON map[string]interface{}
```

Comment:
I would actually decode this to something like:
```go
type header struct {
	Title string `json:"title"`
}
```
as it's all you need to remove a dashboard.